pkg/vpcmodel/explainabilityInput.go

```diff
@@ -46,87 +139,104 @@ func (e *ExplanationArgs) GetConnectionSet() *common.ConnectionSet {
 	return connection
 }
 
-// given src and dst input finds the []nodes they represent
+// given src and dst input and a VPCConfigs finds the []nodes they represent in the config
 // src/dst may refer to:
 // 1. VSI by UID or name; in this case we consider the network interfaces of the VSI
 // 2. Internal IP address or cidr; in this case we consider the vsis in that address range
 // 3. external IP address or cidr
 func (c *VPCConfig) srcDstInputToNodes(srcName, dstName string) (srcNodes, dstNodes []Node,
-	isSrcInternalIP, isDstInternalIP bool, err error) {
-	srcNodes, isSrcInternalIP, err = c.getSrcOrDstInputNode(srcName, "src")
+	isSrcDstInternalIP int, errType int, err error) {
+	var isSrcInternalIP, isDstInternalIP bool
+	srcNodes, isSrcInternalIP, errType, err = c.getSrcOrDstInputNode(srcName, "src")
 	if err != nil {
-		return nil, nil, false, false, err
+		return nil, nil, noInternalIP, errType, err
 	}
-	dstNodes, isDstInternalIP, err = c.getSrcOrDstInputNode(dstName, "dst")
+	dstNodes, isDstInternalIP, errType, err = c.getSrcOrDstInputNode(dstName, "dst")
 	if err != nil {
-		return nil, nil, false, false, err
+		return nil, nil, noInternalIP, errType, err
 	}
 	// only one of src/dst can be external; there could be multiple nodes only if external
 	if !srcNodes[0].IsInternal() && !dstNodes[0].IsInternal() {
-		return nil, nil, false, false, fmt.Errorf("both src %v and dst %v are external", srcName, dstName)
+		return nil, nil, noInternalIP, fatalErr,
+			fmt.Errorf("both src %v and dst %v are external", srcName, dstName)
+	}
+	switch {
+	case isSrcInternalIP && isDstInternalIP:
+		isSrcDstInternalIP = srcAndDstInternalIP
+	case isSrcInternalIP:
+		isSrcDstInternalIP = srcInternalIP
+	case isDstInternalIP:
+		isSrcDstInternalIP = dstInternalIP
+	default:
+		isSrcDstInternalIP = noInternalIP
```

Comment:
this can be captured in a struct type with two fields, instead of converting to enumerated values of 4 options, and then re-converting back to the values per src/dst.

Reply:
It was to begin with. But lint failed due to too many return values. Any other idea how to overcome this?

Reply:
a new struct type that holds these two values as two fields

Reply:
indeed, much better